File: Domain/Book.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Book
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public DateTime Date { get; set; }
        public string Category { get; set; }
        public string Description { get; set; }
        public string Language { get; set; }
        public bool IsCancelled { get; set; }

        public ICollection<BookAttendee> Attendees { get; set; } = new List<BookAttendee>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: client-app/src/app/models/book.ts
import { Profile } from "./profile";

export interface Book {
  id: string;
  title: string;
  author: string;
  date: Date | null;
  category: string;
  description: string;
  language: string;
  hostUsername: string;
  isCancelled: boolean;
  isReading: boolean;
  isHost: boolean;
  host?: Profile;
  attendees: Profile[];
}
export class Book implements Book {
  constructor(init?: BookFormValues) {
    Object.assign(this, init);
  }
}
export class BookFormValues {
  id?: string = undefined;
  title: string = "";
  author: string = "";
  category: string = "";
  description: string = "";
  date: Date | null = null;
  language: string = "";

  constructor(book?: BookFormValues) {
    if (book) {
      this.id = book.id;
      this.title = book.title;
      this.author = book.author;
      this.category = book.category;
      this.description = book.description;
      this.date = book.date;
      this.language = book.language;
    }
  }
}
